### ISSUE TITLE
[Google Search API] no `link` returned for some Ads in Google Search

### ISSUE THREAD

**Issue by marm123** [serpapi/public-roadmap#291]
Some links are missing from Google Search Ads results.

![image](https://user-images.githubusercontent.com/28161162/187159504-dd741613-9d9a-490b-bc3a-e60eb3f9b2db.png)

<a href="https://serpapi.com/playground?q=custom+closets&location=Irvine%2C+California%2C+United+States">Playground</a> | <a href="https://serpapi.com/searches/63090635c999030fee7f5d0d/inspect">Inspect</a>

**Comment by austincollinpena** [comment]
On other tools I've seen this happen because the link is hosted on a Google domain and then redirects to the destination. This is different from a tracking link.

My guess is if you set up a function to follow the link redirect you'll get the actual link.

**Comment by hartator** [comment]
> My guess is if you set up a function to follow the link redirect you'll get the actual link.

We can't do that as this will register a click through Google Ads I would believe.

If there is no way to verify final link in a different way, we would remove that `link` field.

**Comment by austincollinpena** [comment]
That’s an interesting thought.

I wonder what would happen if you stripped any url parameters? Don’t you
think it wouldn’t count as a click?

**Comment by hartator** [comment]
> I wonder what would happen if you stripped any url parameters? Don’t you think it wouldn’t count as a click?

The final link is probably there in clear though. It seems to be just a new format of ad.

**Comment by austincollinpena** [comment]
Is this a won't fix? I get a ton of serp results without links and that negatively affects my application

**Comment by hartator** [comment]
> Is this a won't fix? I get a ton of serp results without links and that negatively affects my application

Yes, this is for now a `wontfix`. Sorry about that.

I am waiting for 2nd set of eyes to double check my findings as I can be wrong.

However, the destination link doesn't seem to be in HTML (or anywhere else) anymore. The only way to get the destination link would be to follow the Google Ads link. Which we won't do as this might be counted as a click. This issue seems not be possible to fix technically on our side unfortunately.

**Comment by austincollinpena** [comment]
Can we test it? I can set up a Google ads campaign where you follow one of these links and we can see if I get a click in the area you searched from.

**Comment by hartator** [comment]
> Can we test it? I can set up a Google ads campaign where you follow one of these links and we can see if I get a click in the area you searched from.

No, because even if this test follow gets ignored there is no guarantee that other follows won't be.
